src/Data/NaturalTransformation.purs

```diff
@@ -0,0 +1,14 @@
+module Data.NaturalTransformation where
+
+-- | A type for natural transformations between functors (although the `Functor`
```

Comment:
I wouldn't mention `Functor` here, and just say "between type constructors" instead. `runCoyoneda` doesn't need this to be a `Functor`, for example.